pymc/distributions/mixture.py

```diff
@@ -398,6 +404,22 @@ def marginal_mixture_logprob(op, values, rng, weights, *components, **kwargs):
     return mix_logp
 
 
+@_get_moment.register(MarginalMixtureRV)
+def get_moment_marginal_mixture(op, rv, rng, weights, *components):
+    ndim_supp = components[0].owner.op.ndim_supp
+    weights = at.shape_padright(weights, ndim_supp)
+
+    if len(components) == 1:
+        moment_components = get_moment(components[0])
+
+    else:
+        moment_components = at.stack(
+            [get_moment(component) for component in components], axis=-ndim_supp - 1
```

Comment:
I would save this in a variable just for readability `mix_axis=-ndim_supp - 1` and use it in the two places. Not super necessary